### DISCUSSION TITLE
Mouse select doesn't work properly in Xorg

### DISCUSSION THREAD

**Discussion by morehockeystats** [microsoft/monaco-editor#3749] [Q&A]
If you select text in Monaco Editor, either by dragging the mouse, or by double-clicking, and then click the middle mouse button, it would not paste the selection. It would paste the buffer selected before interaction with the editor instead - the one that was supposed to be overwritten.

Even when you right-click to "Copy" the selection, the middle-button paste still does not work.

This contradicts the normal behavior under Xorg/X11 window system.

Ubuntu 18.04.4 / Linux 4.15.0-88 / Xorg 7.7 / xdm 1.1.11 / AfterStep 2.2.12 / Firefox 75.0.

**Comment by mattgodbolt** [comment]
I don't know if this is related, but middle-click to paste doesn't work on Firefox (89.0.1) in the same way as described above too.

All interactions I tested with the Xorg selection clipboard worked on Chrome (91.0.4472.114).

**Comment by morehockeystats** [comment]
Here's my update:
it works correctly in Chromium (Chromium: 89.0.4389.90 (Official Build) (64-bit))
it works partially in Firefox (86.0): it pastes what had been selected beforehand, but select to copy itself does not happen).

**Comment by tomfitzhenry** [comment]
I'm not sure how editing/selection is implemented in VS Code/Monaco, but I see other web-based code editors have this same issue:
* https://github.com/codemirror/dev/issues/1097
* https://bugzilla.mozilla.org/show_bug.cgi?id=1567160

In those cases, the difference looks to be explained by Firefox not copying to PRIMARY clipboard on setSelectionRange, as reproducible in https://jsfiddle.net/carter18/7bvh2rpz/. Chrome does write to PRIMARY clipboard in that case.

I'm not familiar with the VS Code codebase, but I see use of setSelectionRange in places: https://github.com/microsoft/vscode/blob/dd4c9da8593873263d96bb2f7d36f4e722cd2808/src/vs/base/browser/ui/inputbox/inputBox.ts#L297
